app/controllers/users_controller.rb

```diff
@@ -1,22 +1,47 @@
 class UsersController < ApplicationController
   before_action :authorize_user
-  before_action :set_user, only: :show
+  before_action :set_user, except: :index
 
 
   def index
     @users = User.all.order(last_sign_in_at: :asc)
   end
 
 
+  def update
+
+    if passwords_match(user_params) && @user.update(user_params)
+      redirect_to @user, notice: t('.success')
+    else
+      flash[:alert] = t('.error') + '  ' + t('.passwords_dont_match')
```

Comment:
This can mask other validation problems.  For instance, when I change the password in the UI, and use a password that is too short (less than 6 chars - see line 157 in `initializers/devise.rb`) then this message (passwords don't match) is misleading.